comment_scan/commander.rb

```diff
@@ -333,6 +334,11 @@ def add(room_id, bot, type, regex, *reason)
             return
         end
 
+        matched_comments = Comment.where(post_type: comment_type).count { |comment| %r{#{regex}}.match(comment.body_markdown.downcase) }
```

Comment:
I think that you may want to do this slightly differently, since `Comment.where(post_type: comment_type)` will give you a *lot* of comments and you only need to check the first 10.

Reply:
We need to check more than just the first 10 (especially if no comments in the whole db match the regex), so i don't really see a better way to do this. I believe it's necessary to run though all the comments. The only optimization I could think of here would be to somehow break when we find the 10th comment that matches (since we don't need to check anymore after that).

Reply:
Yeah, that's what I was thinking. Some kind of `Comment.where(...).each { |comment| count++ if regex_matches; break if count > 10 }`